src/ray/core_worker/test/core_worker_test.cc

```diff
@@ -0,0 +1,880 @@
+
+  std::unique_ptr<ActorHandle> actor_handle;
+
+  // Test creating actor.
+  std::vector<ObjectID> object_ids;
+  std::vector<std::string> object_id_strs;
+  for (int i = 0; i < 2; i++) {
+    object_ids.push_back(ObjectID::FromRandom());
+    object_id_strs.push_back(object_ids.back().Binary());
+  }
+
+  {
+    // Test creating actor.
+    RayFunction func{ray::Language::PYTHON, {c_test_actor_failover_str}};
+
+    std::vector<TaskArg> args;
+    for (const auto &entry : object_id_strs) {
+      auto object_id_buffer = std::make_shared<LocalMemoryBuffer>(
+          reinterpret_cast<uint8_t *>(const_cast<char *>(entry.data())), entry.size());
+      args.emplace_back(TaskArg::PassByValue(object_id_buffer));
+    }
+
+    ActorCreationOptions actor_options{0 /* disable reconstruction */, is_direct_call,
+                                       resources};
+
+    // Create an actor.
+    RAY_CHECK_OK(driver.Tasks().CreateActor(func, args, actor_options, &actor_handle));
+  }
+
+  // Test submitting some tasks with by-value args for that actor.
+  {
+    std::random_device rd;   // Will be used to obtain a seed for the random number engine
+    std::mt19937 gen(rd());  // Standard mersenne_twister_engine seeded with rd()
+    std::uniform_int_distribution<> dis(1, 10);
+    std::uniform_int_distribution<> value_dis(1, 255);
+    const int num_tasks = 3000;
+    const int num_tasks_to_kill_worker = (num_tasks + 1) / 2;
+    std::vector<std::pair<ObjectID, std::vector<uint8_t>>> all_results;
+    for (int i = 0; i < num_tasks; i++) {
+      if (i == num_tasks_to_kill_worker) {
+        RAY_LOG(INFO) << "killing worker";
+        system("pkill mock_worker");
+      }
+
+      // wait for actor being reconstructed.
+      std::vector<uint8_t> arg1(dis(gen), value_dis(gen));
+      auto buffer1 = std::make_shared<LocalMemoryBuffer>(arg1.data(), arg1.size());
+
+      // Create arguments with PassByRef and PassByValue.
+      std::vector<TaskArg> args;
+      args.emplace_back(TaskArg::PassByValue(buffer1));
+
+      TaskOptions options{1, resources};
+      std::vector<ObjectID> return_ids;
+      RayFunction func{ray::Language::PYTHON, {}};
+
+      auto status =
+          driver.Tasks().SubmitActorTask(*actor_handle, func, args, options, &return_ids);
+      if (i < num_tasks_to_kill_worker) {
+        RAY_CHECK_OK(status);
+      }
+
+      RAY_CHECK(return_ids.size() == 1);
+      all_results.emplace_back(std::make_pair(return_ids[0], arg1));
+    }
+
+    for (int i = 0; i < num_tasks; i++) {
+      const auto &entry = all_results[i];
+      std::vector<ObjectID> return_ids;
+      return_ids.push_back(entry.first);
+      std::vector<std::shared_ptr<RayObject>> results;
+      RAY_CHECK_OK(driver.Objects().Get(return_ids, -1, &results));
+      RAY_CHECK(results.size() == 1);
+
+      if (results[0]->GetMetadata() != nullptr && results[0]->GetMetadata()->Size() > 0) {
```

Comment:
1) we should also check the if block only occurs for `i > num_tasks_to_kill_worker` and the else block for the opposite condition.
2) Add a `RayObject::HasMetadata` method to simplify this check.

Reply:
1. This is not accurate, as there are delays before kill is called, and the process is actually killed, thus it might be possible that there are a few tasks after `num_tasks_to_kill_worker` still return success.
2. OK, will do.